dart/lib/src/sentry.dart

```diff
@@ -55,13 +55,23 @@ class Sentry {
     AppRunner appRunner,
     List<Integration> initialIntegrations,
   ) async {
-    // if no environment is set, we set 'production' by default, but if we know it's
-    // a non-release build, or the SENTRY_ENVIRONMENT is set, we read from it.
-    if (const bool.hasEnvironment('SENTRY_ENVIRONMENT') || !isReleaseMode) {
+    // We infer the enviroment based on the release/non-release and profile
+    // constants.
+    const environment = isReleaseMode
+        ? defaultEnvironment
+        : isProfileMode
+            ? 'profile'
+            : 'debug';
+
+    // If the environment variable is set, we read it and fallback to the
+    // previously infered constant if it could not be read.
+    if (const bool.hasEnvironment('SENTRY_ENVIRONMENT')) {
       options.environment = const String.fromEnvironment(
         'SENTRY_ENVIRONMENT',
-        defaultValue: 'debug',
+        defaultValue: environment,
       );
+    } else {
+      options.environment = environment;
     }
```

Comment:
we can probably leave it without the `defaultEnvironment` setting here because of:

https://github.com/getsentry/sentry-dart/blob/a9979b0adc18b2545ad907ebae8b1f4b558844d6/dart/lib/src/sentry_client.dart#L101-L102

if nothing is set, we automatically set the `defaultEnvironment` when preparing the event, so either we remove the `defaultEnvironment` from here, or we can just remove the fallback from the `_prepareEvent` method, either way is fine for me, but we could remove the business logic duplication, wdyt?

Reply:
I'd rather set is sooner than later, as we have the correct environment in the options for the whole lifecycle. Will remove the fallback, thank you!